base/base/bit_cast.h

```diff
@@ -13,6 +14,12 @@ template <typename To, typename From>
 std::decay_t<To> bit_cast(const From & from)
 {
     To res {};
-    memcpy(static_cast<void*>(&res), &from, std::min(sizeof(res), sizeof(from)));
+    if constexpr (std::endian::native == std::endian::little)
+      memcpy(static_cast<void*>(&res), &from, std::min(sizeof(res), sizeof(from)));
+    else
+    {
+      uint32_t offset = (sizeof(res) > sizeof(from)) ? (sizeof(res) - sizeof(from)) : 0;
+      memcpy(reinterpret_cast<char *>(&res) + offset, &from, std::min(sizeof(res), sizeof(from)));
+    }
```

Comment:
For me it seems it would be better to do the conversion "LE -> native" during deserialization while we're reading data from disk.